### PR TITLE
Make Network Rest Great Again

### PR DESCRIPTION
**Pull request by Hiro-Nakamura**
We had certain dependencies on the client of depending on the Socket Updates to properly update our local client data.  

however, this won't display the application correctly when we set out Network = "rest".  So these changes should enable a local client app to properly update itself without the input of socket updates.

## Release Notes

- [fix] have the client trigger local datacollection update handlers when we modify data.
- [fix] patch additional ABFieldUser quirks when working with values that need .id = .username
- [sync] with appropriate changes to core

### CODE REVIEW

File: AppBuilder/platform/ABModel.js [hunk 1]
```diff
@@ -49,12 +49,14 @@ function errorPopup(error) {
...
    // update event with this data.
-   if (model.AB.Network.type() != "socket") {
-      model.AB.emit(key, {
-         objectId: model.object.id,
-         data,
-      });
-   }
+   // ## NOTE: attempting to see performance difference
+   // ## if we reduce our socket updates
+   // if (model.AB.Network.type() != "socket") {
```
Comment (nh758) [inline AppBuilder/platform/ABModel.js:54]: I don't understand this. Don't we still want this check?  Won't this cause duplicate updates when we use sockets?

**Review by nh758 — CHANGES_REQUESTED** [review]
One question below. Otherwise looks good, but again want to see our e2e test passing.

**Review by nh758 — APPROVED** [review]
Looks good

**Review by wongpratan — APPROVED** [review]

**Comment by Hiro-Nakamura** [comment]
How do I get this to kick off the KitchenSink e2e tests?

**Pull request merged**